index.js

```diff
@@ -146,45 +191,58 @@ function nowServing(line) {
 const line = []
 
 // ***** Scope & Closures - Question 1 *****
+let lineCounter = 1
 
+function takeATicketNumber(line) {
+  line.push(lineCounter)
+  lineCounter++
+  return `Welcome. You are ticket number ${lineCounter-1}`
+}
 
 // *** Uncomment the lines below to test
-// console.log("%cScope & Closures - Question 1", "color: red")
+console.log("%cScope & Closures - Question 1", "color: red")
 
-// console.log(takeATicketNumber(line))
-// // => `Welcome. You are ticket number 1`
+console.log(takeATicketNumber(line))
+// => `Welcome. You are ticket number 1`
 
-// console.log(takeATicketNumber(line))
-// // => `Welcome. You are ticket number 2`
+console.log(takeATicketNumber(line))
+// => `Welcome. You are ticket number 2`
 
-// console.log(nowServing(line))
-// // => `Currently serving 1.`
+console.log(nowServing(line))
+// => `Currently serving 1.`
 
-// console.log(nowServing(line))
-// // => `Currently serving 2.`
-// console.log("%c----------", "color: red") 
+console.log(nowServing(line))
+// => `Currently serving 2.`
 
-// console.log(takeATicketNumber(line))
-// // => `Welcome. You are ticket number 3`
+console.log(takeATicketNumber(line))
+// => `Welcome. You are ticket number 3`
 
+console.log("%c----------", "color: red") 
 
-// ***** Scope & Closures - Question 2 *****
 
 
+// ***** Scope & Closures - Question 2 *****
+function ticketNumberGeneratorFunc() {
+  lineCounter = 1
+  return takeATicketNumber
+}
```

Comment:
This won't quite work as intended - in this example, you're still using the `lineCounter` global variable. The way to get this working with a closure would be something like this:

```js

function ticketNumberGeneratorFunc() {

  let ticket = 0
  function takeATicketNumber(line) {
    ticket++
    line.push(ticket)
    return `Welcome. You are ticket number ${ticket}`
  }

  return takeATicketNumber
}
```

By creating a new function inside the `ticketNumberGeneratorFunc` and returning it, that function keeps access to the new `ticket` variable inside the outer function.